### DISCUSSION TITLE
DatePicker with disabled dates retains :hover classes for out of bound dates

### DISCUSSION THREAD

**Discussion by adi-unni** [jpmorganchase/salt-ds#3913] [Q&A] [answered]
Hey Team,

Not sure if this is a feature or a bug in the [DatePicker with disabled dates component](https://www.saltdesignsystem.com/salt/components/date-picker/examples#with-disabled-dates). This is a bit of an edge case but when hovering over the current months dates that are disabled, say a weekend in July, the date stays greyed out and its not an interactive icon. However, say a previous month or a next month (with the .saltCalendarDay-outOfRange class) also falls within that disabled category, it still gains the :hover class making it coloured black and seemingly interactive as if you can jump to the next month.

Actual:
![Screenshot 2024-07-31 at 12 00 44](https://github.com/user-attachments/assets/d094fcd9-9688-4c06-a677-5b99dd6f91c2)
Expected:
![Screenshot 2024-07-31 at 11 59 25](https://github.com/user-attachments/assets/6e01610e-cf8b-4fd1-8b2a-f6ed92053a54)

Think its just this style getting overridden if it is a bug
![Screenshot 2024-07-31 at 12 01 08](https://github.com/user-attachments/assets/ed4f8990-66bb-4a22-84e7-2cd67da81534)

**Comment by joshwooding** [comment, marked as answer]
Hey, that looks to be a bug with the component. I've created an issue to track it. Thanks for reporting it!
